source/io/module_javaScriptInvoke.js

```diff
@@ -338,6 +324,19 @@
             };
         };
 
+        var createJavaScriptParametersArray = function (isInternalFunction, parametersPointer, parametersCount) {
+            var parameters = new Array(parametersCount);
+            for (var index = 0; index < parametersCount; index += 1) {
+                var parameterValueRef = createValueRefFromPointerArray(parametersPointer, index);
+                if (!isInternalFunction) {
```

Comment:
I was wondering if we need to pass a ValueRef to all internal functions present and future. I think it is OK. It will make for a more efficient API for internal functions.

Reply:
I hope eventually the concept for internal functions only means that we look at a private list of functions and does not involve a change in behavior like it does now.

Eventually each parameter should be individually configurable as being passed as a valueRef and the user should have access to all the advanced jsapi functions.